collector/mesos_agent/mesos_agent.go

```diff
@@ -115,36 +149,109 @@ type resourceStatistics struct {
 	NetTxDropped uint64 `json:"net_tx_dropped,omitempty"`
 }
 
+func (h *MesosAgentCollector) RunPoller() {
+	ticker := time.NewTicker(h.PollPeriod * time.Second)
+
+	// Poll once immediately
+	h.pollMesosAgent()
+	for _, m := range h.transformContainerMetrics() {
+		h.MetricsChan <- m
+	}
+
+	for {
+		select {
+		case _ = <-ticker.C:
+			h.pollMesosAgent()
+			for _, m := range h.transformContainerMetrics() {
+				h.MetricsChan <- m
+			}
+		}
+	}
+}
```

Comment:
if you need to use channels, because in your next refactor you want to interrupt the waiting time by an event, you can do
```go
  select {
    case <-ticker.C:
  }
```